tests/test_core.py

```diff
@@ -89,6 +89,18 @@ def test_read_grid_from_input_should_set_grid_attr_when_called(simple_sim):
     assert simple_sim.grid.r_min == 0
     assert simple_sim.grid.r_max == 1
 
+def test_prepare_simulation_physics_modules(simple_sim):
+    """Tests that prepare_simulation correctly initializes the physics_modules"""
+    other_sim = simple_sim
```

Comment:
I'm not sure that this does what you hope it does. Since `simple_sim` is an object, I think this is just giving a new "name" to the same object. So when you make the calls below to `other_sim`, it is also changing `simple_sim`. You can use the `id` function to see if these both point to the same underlying object.